inductiva/stellarators/scenarios/coils/coils.py

```diff
@@ -185,8 +189,40 @@ def from_curves_file(cls,
 
         return cls(coils, num_field_periods)
 
-    def simulate(self):
-        pass
+    def simulate(
+        self,
+        simulator: Simulator = Simsopt(),
+        resource_pool_id: Optional[UUID] = None,
+        run_async: bool = False,
+        magnetic_field_filename: str = 'magnetic_field.npz',
+        plasma_surface_filename: str = 'input.QA',
+    ) -> tasks.Task:
+        """Simulates the scenario.
+
+        Args:
+            simulator: The simulator to use for the simulation.
+            resource_pool_id: The resource pool to use for the simulation.
+        """
+        coil_coefficients = [coil.curve_coefficients for coil in self.coils]
+        coil_currents = np.array([coil.current for coil in self.coils])
+
+        coil_coefficients_filename = 'coil_coefficients.npz'
+        coil_currents_filename = 'coil_currents.npz'
+
+        np.savez(coil_coefficients_filename, *coil_coefficients)
+        np.savez(coil_currents_filename, coil_currents)
```

Comment:
Move this to a method `create_input_files`. See other scenarios to see exactly how this should be done.